Fix: resourceToWeightMap will never be nil here

kerthcet opened the pull request:
Signed-off-by: kerthcet <kerthcet@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
/sig scheduling

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:
`Len()` is better to tell whether a map is empty or not, and `resourceToWeightMap` will never be nil here. See https://github.com/kubernetes/kubernetes/blob/7d9c0e0a78e519cac0f892a8be2f063bedc94bad/pkg/scheduler/framework/plugins/noderesources/resource_allocation.go#L141
#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

denkensk commented on pkg/scheduler/framework/plugins/noderesources/resource_allocation.go:
```diff
@@ -51,7 +51,7 @@ func (r *resourceAllocationScorer) score(
 		return 0, framework.NewStatus(framework.Error, "node not found")
 	}
-	if r.resourceToWeightMap == nil {
+	if len(r.resourceToWeightMap) == 0 {
```
Just a personal opinion：I think this is a way to protect the robustness of the code. Although it’s initialized before in all the places to call this function. But it's not mandatory.

kerthcet replied:
We have defaults for nil resources, so actually this will not happen I think.

ahg-g commented on pkg/scheduler/framework/plugins/noderesources/most_allocated_test.go:
```diff
@@ -93,20 +93,6 @@ func TestMostAllocatedScoringStrategy(t *testing.T) {
 			expectedScores: []framework.NodeScore{{Name: "node1", Score: 62}, {Name: "node2", Score: 50}},
 			resources:      defaultResources,
 		},
-		{
```
I would keep the tests

kerthcet replied:
resources is nil is not allowed, test will fail then.

ahg-g replied:
Failing on empty/nil map isn't new and the tests already existed and were passing. Specifically, I would like to continue to have a test that ensures we don't crash if the map was nil/empty

kerthcet replied:
Make sense, add back the tests.

kerthcet committed:
Fix: resourceToWeightMap will never be nil

Use len() instead of telling whether it's nil
Add tests to guarantee that when resourceToWeightMap is nil, scheduler will not crash

Signed-off-by: kerthcet <kerthcet@gmail.com>

kerthcet commented:
/assign @ahg-g 
for approval

kerthcet commented:
Kindly ping @ahg-g

ahg-g commented:
/lgtm
/approve
